### PR TITLE
Husky pre commit hooks

### PR DESCRIPTION
**Pull request by webb04**
## Changes
- Run npm lint before allowing commits
- Only run lint in `src` folder

This can be extended to run tests before commits but we might need to speed them up first (hopefully the new testing library helps)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,14 +8,15 @@
...
     "upload": "node-riffraff-artifact",
     "copy-manifest": "cp dist/assets/manifest.json dist/server/manifest.json",
     "copy-fonts": "cp -R assets/fonts dist/assets/fonts",
     "test": "jest --config config/jest.config.js --verbose",
     "test:coverage": "npm run test; open coverage/index.html",
     "test:watch": "jest --config config/jest.config.js --watch",
-    "lint": "node_modules/eslint/bin/eslint.js -c config/.eslintrc.js ./ --ext '.ts,.tsx'",
-    "lint:fix": "node_modules/eslint/bin/eslint.js -c config/.eslintrc.js ./ --ext '.ts,.tsx' --fix",
+    "lint": "node_modules/eslint/bin/eslint.js -c config/.eslintrc.js ./src --ext '.ts,.tsx'",
+    "lint:fix": "node_modules/eslint/bin/eslint.js -c config/.eslintrc.js ./src --ext '.ts,.tsx' --fix",
```
Comment (JamieB-gu) [inline package.json:19]: Does this mean we'll no longer lint `webpack.config.ts` for example?

**Review by JamieB-gu — COMMENTED** [review]
Couple of questions:

1. Would you be able to discuss a bit further what Husky does and what you think the benefits are?
2. Is there a straightforward way to disable it on a developer-by-developer basis, in case some devs want to opt out either temporarily or permanently?

**Review by dskamiotis — APPROVED** [review]

**Comment by webb04** [comment]
> Couple of questions:
> 
> 1. Would you be able to discuss a bit further what Husky does and what you think the benefits are?
> 2. Is there a straightforward way to disable it on a developer-by-developer basis, in case some devs want to opt out either temporarily or permanently?

Quite often I'll commit and only find the lint errors when checking GitHub. I've had to make many `lint fixes` commits which can sometimes clutter the commit history. I was hoping to have less of these reach GitHub and they also don't have to be part of a peer review. 

`export HUSKY=0` should turn it off https://typicode.github.io/husky/#/?id=disable-hooks-in-ci

**Comment by dskamiotis** [comment]
> > Couple of questions:
> > 
> > 1. Would you be able to discuss a bit further what Husky does and what you think the benefits are?
> > 2. Is there a straightforward way to disable it on a developer-by-developer basis, in case some devs want to opt out either temporarily or permanently?
> 
> Quite often I'll commit and only find the lint errors when checking GitHub. I've had to make many `lint fixes` commits which can sometimes clutter the commit history. I was hoping to have less of these reach GitHub and they also don't have to be part of a peer review.
> 
> `export HUSKY=0` should turn it off https://typicode.github.io/husky/#/?id=disable-hooks-in-ci

Totally agree with this @webb04 I also do this and find cluttering

**Review by frankie297 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -102,7 +103,9 @@
     "eslint-plugin-jsx-a11y": "^6.3.1",
     "eslint-plugin-react": "^7.21.4",
     "html-webpack-plugin": "^4.5.0",
+    "husky": "^5.0.0-alpha.6",
     "jest": "^26.5.3",
+    "pinst": "^2.0.0",
```
Comment (JamieB-gu) [inline package.json:108]: @webb04 given [this section](https://github.com/typicode/husky/tree/master#ci-servers) in the Husky documentation, do we need `pinst`?

**Pull request merged**